sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -452,6 +452,26 @@ case class OuterReference(e: NamedExpression)
   final override val nodePatterns: Seq[TreePattern] = Seq(OUTER_REFERENCE)
 }
 
+/**
+ * A place holder used to hold a reference that has been resolved to a field outside of the current
+ * plan. And the reference cannot be resolved by the immediate outer plan of the current plan.
+ * It is used only for queries containing nested correlations and are used only in
```

Comment:
"It is used only.. and are used only" -> "We use it only for queries containing nested correlation and only in SubqueryExpression..."